WooCommerce/Classes/Notifications/PushNotificationsManager.swift

```diff
@@ -121,10 +124,30 @@ extension PushNotificationsManager {
 
     /// Resets the Badge Count.
     ///
-    func resetBadgeCount() {
-        configuration.application.applicationIconBadgeNumber = 0
+    func resetBadgeCount(type: Note.Kind) {
+        guard let siteID = siteID else {
+            return
+        }
+        let action = NotificationCountAction.resetNotificationCount(siteID: siteID, type: type) { [weak self] in
+            self?.loadNotificationCountAndUpdateApplicationBadgeNumberAndPostNotifications(siteID: siteID, type: type)
+        }
+        ServiceLocator.stores.dispatch(action)
```

Comment:
⚠️  Same as above, we don't need to use `ServiceLocator` since there is an injected `configuration`.

Reply:
similar to above, updated in https://github.com/woocommerce/woocommerce-ios/pull/2320/commits/7620bb0da4f901ebfd141939e5511a9ab657c39e